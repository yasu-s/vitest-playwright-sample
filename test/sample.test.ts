import { afterAll, beforeAll, describe, test, expect } from 'vitest'
import { preview, PreviewServer } from 'vite'
import { chromium, Browser, Page } from 'playwright'
import { expect as expectEx } from '@playwright/test'
import { toMatchImageSnapshot } from 'jest-image-snapshot'

declare global {
  namespace jest {
    interface Matchers<R> {
      toMatchImageSnapshot(): R
    }
  }
}

expect.extend({ toMatchImageSnapshot })

describe('sample', async () => {
  let server: PreviewServer
  let browser: Browser
  let page: Page

  beforeAll(async () => {
    server = await preview({ preview: { port: 3000 }, build: { outDir: 'public' } })
    browser = await chromium.launch()
    page = await browser.newPage()
  })

  afterAll(async () => {
    await browser.close()
    await new Promise<void>((resolve, reject) => {
      server.httpServer.close(error => error ? reject(error) : resolve())
    })
  })

  test('title', async () => {
    // setup
    await page.goto('http://localhost:3000')

    // verify
    const title = await page.title()
    expect(await page.content()).toMatchSnapshot()
    expect(await page.screenshot()).toMatchImageSnapshot()
    expect(title).toBe('Vitest Test Page')
  })

  test('pagetitle', async () => {
    // setup
    await page.goto('http://localhost:3000')

    // verify
    const pagetitle = page.locator('#pagetitle')
    expect(await pagetitle.innerText()).toBe('Vitest Test')
    await expectEx(pagetitle).toHaveText('Vitest Test')
    expect(await page.screenshot()).toMatchImageSnapshot()
  })
})
