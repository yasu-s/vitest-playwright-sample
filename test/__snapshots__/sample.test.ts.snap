// Vitest Snapshot v1

exports[`sample > title > aaa 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <title>Vitest Test Page</title>
</head>
<body>
  <div id=\\"pagetitle\\">
    Vitest Test
  </div>
  <div>
    <input type=\\"text\\" id=\\"test\\">
  </div>


</body></html>"
`;
